### PR TITLE
Security Fix: Add checksum verification of dynamic asm downloads

### LINKED ISSUE #876: JCOV Tools Download Without Integrity Checks
Found as part of TrailOfBits Audit:

Ref: TOB-TEMURIN-12

The jcov.sh script downloads ASM tools without verifying their integrity or authenticity:  
 

The integrity or authenticity should be verified using a hashsum like SHA256, or a signature  
like a GPG signature. This ensures that the system is receiving the same data across  
multiple downloads. This download does use HTTPS, so this issue is marked as low severity.

### PR DESCRIPTION
**Pull request by steelhead31**
Fixes #876 

Include MD5 checksum verification of ASM downloads.

### CODE REVIEW

File: tools/code-tools/jcov.sh [hunk 1]
```diff
@@ -154,20 +164,20 @@ pushd $REPO_DIR
...
-    tar -czf ../../$main_file-$tip_shortened.tar.gz *.jar readme.txt
-  popd
-  echo "Manually renaming $main_file-$tip_shortened.tar.gz as $main_file-tip..tar.gz to provide latest-unstable-recommended file"
-  ln -fv $main_file-$tip_shortened.tar.gz $main_file-tip.tar.gz
-  cleanRepo
-
-  echo "Resetting repo back to master"
-  resetRepo master
-  hashArtifacts
+  # pushd build
```
Comment (karianna) [inline tools/code-tools/jcov.sh:167]: commented out code?

**Review by karianna — CHANGES_REQUESTED** [review]

File: tools/code-tools/jcov.sh (hunk 1 above)
Comment (steelhead31) [reply tools/code-tools/jcov.sh]: Comments removed, code re-enabled.. nice spot.

File: tools/code-tools/jcov.sh [hunk 2]
```diff
@@ -72,9 +72,19 @@ function getAsmDeps() {
     local tool_prop="`echo $tool|sed "s/-/./g"`.jar"
     local tool_versioned="$tool-$asm_version.jar"
     local tool_url="$main_url/$tool/$asm_version/$tool_versioned"
+    local tool_checksum_url="$tool_url.md5"
     if [ "$asm_manual" == "true" ] ; then
       if [ ! -e $tool_versioned ] ; then
+        wget $tool_checksum_url
         wget $tool_url
+        check_md5sum=`md5sum $tool_versioned | cut -d" " -f1`
+        check_file=`cat $tool_versioned.md5`
```
Comment (sxa) [inline tools/code-tools/jcov.sh:81]: Can this be done with `md5sum -c` on the `md5` file for simplicity?

**Review by sxa — COMMENTED** [review]
Simplicity suggestion

File: tools/code-tools/jcov.sh (hunk 2 above)
Comment (steelhead31) [reply tools/code-tools/jcov.sh:81]: Sadly it isnt, the md5 from the maven repo, only contains the checksum, and not the checksum followed by the filename, required by md5sum -c ... le sigh.

Comment (sxa) [reply tools/code-tools/jcov.sh:81]: Unfortunate - ok I'll approve for now

**Review by sxa — APPROVED** [review]
Approving, but as per earlier comment let's get another issue created to look at doing this in a better way in the future - and to look for other similar instances

**Comment by Haroon-Khel** [comment]
ping @karianna for rereview

**Review by Haroon-Khel — APPROVED** [review]

**Review by karianna — APPROVED** [review]

**Pull request merged**